addon/components/materialize-loader.js

```diff
@@ -0,0 +1,52 @@
+import Ember from 'ember';
+import layout from '../templates/components/materialize-loader';
+
+export default Ember.Component.extend({
+  layout: layout,
+  mode: 'indeterminate',
+  percent: 0,
+  size: 'big',
+  active: true,
+  color: null,
+  classNameBindings: ['isBarType:progress:preloader-wrapper', 'active:active', 'size'],
+
+  isBarType: function () {
+    return ['determinate', 'indeterminate'].indexOf(this.get('mode')) >= 0;
+  }.property('mode'),
+
+  isDeterminate: function () {
+    return ['determinate'].indexOf(this.get('mode'));
+  }.property('mode'),
+
+  barStyle: function () {
+    if (this.get('mode') === 'determinate') {
+      var pct = this.get('percent');
+      return `width: ${pct}%`;
```

Comment:
Ember-cli crashes at this point. I'm getting the following error message:

SyntaxError: illegal character
return `width: ${pct}%`;